bigframes/session/executor.py

```diff
@@ -218,7 +274,7 @@ def _run_execute_query(
             else:
                 raise
 
-    def _with_cached_executions(self, node: nodes.BigFrameNode) -> nodes.BigFrameNode:
+    def _get_optimized_plan(self, node: nodes.BigFrameNode) -> nodes.BigFrameNode:
```

Comment:
Could we get a docstring for this? Or maybe rename? I'd love to understand when we'd use this? It seems we'd always want to use an "optimized" plan, as no downsides are documented here.

Reply:
added docstring. only optimization right now is to apply caching. no real downsides at this stage, other than mutating the structure of the tree, which is fine at execution stage (don't want to do it earlier as it would interfere with implicit joining).